### PR TITLE
Split out imodel transformer

### PR DESCRIPTION
**Pull request by MichaelBelousov**
- remove imodeljs transformer APIs from `@bentley/imodeljs-backend` package and instead put them in a new `@bentley/imodeljs-transformer`
- move non-transformer related test utilities in `IModelTransformerUtils.ts` into `IModelTestUtils.ts` and make imodel transformer tests import those utilities from `@bentley/imodeljs-backend/lib/test/*`
- make `IModelTestUtils` a namespace instead of a class of static methods, `IModelTransformerUtils` was a namespace. This can be flipped by preference.
- move *iModelTransformations and Data Exchange* article to its own transformer section

### CODE REVIEW

File: core/backend/src/test/IModelTestUtils.ts [hunk 1]
```diff
@@ -575,17 +588,1160 @@ export class IModelTestUtils {
...
+    informationRecord2.sourceString = `${informationRecord2.sourceString}-Updated`;
+    informationRecord2.update();
+    // delete InformationRecord3
+    const informationRecodeCode3: Code = new Code({ spec: informationRecordCodeSpec.id, scope: informationModelId, value: "InformationRecord3" });
+    const informationRecordId3: Id64String = sourceDb.elements.queryElementIdByCode(informationRecodeCode3)!;
+    assert.isTrue(Id64.isValidId64(informationRecordId3));
+    sourceDb.elements.deleteElement(informationRecordId3);
+  }
+
+  export async function prepareTargetDb(targetDb: IModelDb): Promise<void> {
```
Comment (calebmshafer) [inline core/backend/src/test/IModelTestUtils.ts:1033]: These methods don't really seem general enough to be in here... What was the reasoning behind moving them out of the TransformationUtils into this file?

Comment (MichaelBelousov) [reply core/backend/src/test/IModelTestUtils.ts:1033]: yeah, with a name like `prepareTargetDb`, it is somewhat transformer specific. There are some plain backend tests (in `DeleteDefinitionElements.test.ts` that use `prepareSourceDb` even not in a transformation related test, which motivated me to move everything that didn't directly depend on the transformation package. I don't think I should move those tests out of the backend package, maybe renaming them to `prepareIModelScenario1` is better? and then possibly change the `prepareTargetDb` to `prepareIModelScenario2` and put that in the transformer package's personal test utilities?

Comment (MichaelBelousov) [reply core/backend/src/test/IModelTestUtils.ts:1033]: do you have any alternatives?

Comment (calebmshafer) [reply core/backend/src/test/IModelTestUtils.ts:1032]: If it is going to stay here, ideally something more specifc to what kind of iModel it is creating. I haven't reviewed all the code for creation but I assume it's building up an iModel that stresses some specific piece. Knowing what that piece is would be helpful for anyone going to use it.

Comment (MichaelBelousov) [reply core/backend/src/test/IModelTestUtils.ts]: I chose `ExtensiveTestScenario`, it is a non-exhaustive amalgam of various elements and schema items for testing different schema and element processing cases, for instance, the transform tests use it to test converting elements between classes, among other things. While originally for testing certain transformer cases, it is used in the DeleteDefinitionElements test and I would consider it more of a non-exhaustive collection of interesting cases. I can't think of a more specific name myself, unless we split the model into several smaller models. I also think it's valuable having a feature-exhaustive iModel to run tests on, to hit more edge cases, so I vote we don't split it.

File: test-apps/imodel-transformer/src/Main.ts [hunk 2]
```diff
@@ -146,8 +147,8 @@ void (async () => {
         assert(!(args.sourceStartChangesetIndex && args.sourceStartChangesetId), "Pick single way to specify starting changeset");
         if (args.sourceStartChangesetIndex) {
           args.sourceStartChangesetId = await IModelHubUtils.queryChangesetId(requestContext, sourceIModelId, args.sourceStartChangesetIndex);
-        } else {
-          args.sourceStartChangesetIndex = await IModelHubUtils.queryChangesetIndex(requestContext, sourceIModelId, args.sourceStartChangesetId!);
+        } else if (args.sourceStartChangesetId) {
```
Comment (ColinKerr) [inline test-apps/imodel-transformer/src/Main.ts:150]: I was not expecting any logic changes in this PR.  Is this a mistake?  If this is an intentional change in behavior is it tested?

Comment (MichaelBelousov) [reply test-apps/imodel-transformer/src/Main.ts:150]: I will try removing it, maybe it was transient while I was moving stuff around, but I had a typescript error where it couldn't infer that things had been properly set, so I changed the code to be explicit.

Comment (MichaelBelousov) [reply test-apps/imodel-transformer/src/Main.ts:150]: alright, it looks like my mistake is that my linter removed an "unnecessary assertion" operator (the `!`) while my types were invalid (implicitly `any`), and I didn't catch that while combing the diff, so I added the condition to the "else" since it was logically redundant. I'll just add the `!` back and remove this change

**Review by ColinKerr — APPROVED** [review]

File: core/transformer/package.json [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+    "eslint": "^7.11.0",
+    "mocha": "^8.3.2",
+    "npm-run-all": "^4.1.5",
+    "nyc": "^15.1.0",
+    "rimraf": "^3.0.2",
+    "sinon": "^9.0.2",
+    "typescript": "~4.3.0"
+  },
+  "dependencies": {
+    "@bentley/imodeljs-native": "3.0.6",
```
Comment (calebmshafer) [inline core/transformer/package.json:79]: Why do you need to take an explicit dependency on `@bentley/imodeljs-native`? Is something missing from the backend that should be re-exported?

Comment (MichaelBelousov) [reply core/transformer/package.json:79]: `IModelExporter.ts` uses `IModelJsNative.ChangedInstanceOpsProps` and `IModelJsNative.ChangedInstanceIdsProps`, if there's a place to re-export types from imodeljs-native from backend, I can add them. Since they're just types it's also possible to get around having a dependency in other ways, but I assume re-exporting from the backend package is better?

Comment (calebmshafer) [reply core/transformer/package.json:79]: You should be able to import them directly from the barrel file since they are both re-exported, `IModelJsNative` and `NativeLoggerCategory`.  https://github.com/imodeljs/imodeljs/blob/master/core/backend/src/imodeljs-backend.ts#L41

Comment (MichaelBelousov) [reply core/transformer/package.json]: done

**Comment by MichaelBelousov** [comment]
@mergifyio update
